src/Tribe/Meta/Chunker.php

```diff
@@ -0,0 +1,811 @@
+	 *
+	 * @param int    $object_id
+	 * @param string $meta_key
+	 *
+	 * @return bool
+	 */
+	protected function applies( $object_id, $meta_key ) {
+		$applies = ! $this->is_chunker_logic_meta_key( $meta_key )
+		           && $this->is_supported_post_type( $object_id )
+		           && ( empty( $meta_key ) || $this->is_chunkable( $object_id, $meta_key ) );
+
+		return $applies;
+	}
+
+	/**
+	 * Whether the meta key is one used by the chunker to keep track of its operations or not.
+	 *
+	 * @param string $meta_key
+	 *
+	 * @return bool
+	 */
+	protected function is_chunker_logic_meta_key( $meta_key ) {
+		return 0 === strpos( $meta_key, $this->meta_key_prefix );
+	}
+
+	/**
+	 * Whether a post ID and meta key couple is registered as chunkable or not.
+	 *
+	 * @param int    $post_id
+	 * @param string $meta_key
+	 *
+	 * @return bool
+	 */
+	public function is_chunkable( $post_id, $meta_key ) {
+		$key = $this->get_key( $post_id, $meta_key );
+
+		return array_key_exists( $key, $this->chunks_cache );
+	}
+
+	/**
+	 * Deletes all the chunks for a post ID and meta key couple.
+	 *
+	 * @param int    $object_id
+	 * @param string $meta_key
+	 */
+	protected function delete_chunks( $object_id, $meta_key ) {
+		/** @var wpdb $wpdb */
+		global $wpdb;
+		$chunk_meta_key = $this->get_chunk_meta_key( $meta_key );
+		$delete = "DELETE FROM {$wpdb->postmeta} WHERE (meta_key = %s OR meta_key = %s) AND post_id = %d";
+		$wpdb->query( $wpdb->prepare( $delete, $chunk_meta_key, $meta_key, $object_id ) );
+	}
+
+	/**
+	 * Returns the meta key used to indicate a chunk for a meta key.
+	 *
+	 * @param string $meta_key
+	 *
+	 * @return string
+	 */
+	public function get_chunk_meta_key( $meta_key ) {
+		return $this->get_chunkable_meta_key( $meta_key ) . '_chunk';
+	}
+
+	/**
+	 * Removes the checksum used to verify the integrity of the chunked values.
+	 *
+	 * @param int    $object_id
+	 * @param string $meta_key
+	 */
+	protected function remove_checksum_for( $object_id, $meta_key ) {
+		/** @var wpdb $wpdb */
+		global $wpdb;
+		$data = array(
+			'post_id'  => $object_id,
+			'meta_key' => $this->get_checksum_key( $meta_key ),
+		);
+		$wpdb->delete( $wpdb->postmeta, $data );
+	}
+
+	/**
+	 * Returns the meta_key used to store the chunked meta checksum for a specified meta key.
+	 *
+	 * @param string $meta_key
+	 *
+	 * @return string
+	 */
+	public function get_checksum_key( $meta_key ) {
+		return $this->meta_key_prefix . $meta_key . '_checksum';
+	}
+
+	/**
+	 * Whether a value should be chunked or not.
+	 *
+	 * @param int    $post_id
+	 * @param string $meta_key
+	 * @param mixed  $meta_value
+	 *
+	 * @return bool
+	 */
+	public function should_be_chunked( $post_id, $meta_key, $meta_value ) {
+		$should_be_chunked = false;
+
+		$max_allowed_packet = $this->get_max_chunk_size();
+		$serialized = maybe_serialize( $meta_value );
+		$byte_size = $this->get_byte_size( $serialized );
+		if ( $byte_size > .8 * $max_allowed_packet ) {
```

Comment:
Why `.8`? I bet there's a good reason, but it _could_ be useful to add a one liner comment with a hint for future maintainers.